examples/tv-casting-app/tv-casting-common/core/CommissionerDeclarationHandler.cpp

```diff
@@ -38,39 +38,48 @@ CommissionerDeclarationHandler * CommissionerDeclarationHandler::GetInstance()
     return sCommissionerDeclarationHandler_;
 }
 
-// TODO: In the following PRs. Implement setHandler() for CommissionerDeclaration messages and expose messages to higher layers for
-// Linux, Android and iOS.
 void CommissionerDeclarationHandler::OnCommissionerDeclarationMessage(
     const chip::Transport::PeerAddress & source, chip::Protocols::UserDirectedCommissioning::CommissionerDeclaration cd)
 {
-    ChipLogProgress(AppServer,
-                    "CommissionerDeclarationHandler::OnCommissionerDeclarationMessage(), calling CloseCommissioningWindow()");
-    // Close the commissioning window. Since we recived a CommissionerDeclaration message from the Commissioner, we know that
-    // commissioning via AccountLogin cluster failed. We will open a new commissioningWindow prior to sending the next
-    // IdentificationDeclaration Message to the Commissioner.
-    chip::Server::GetInstance().GetCommissioningWindowManager().CloseCommissioningWindow();
-    support::ChipDeviceEventHandler::SetUdcStatus(false);
+    ChipLogProgress(AppServer, "CommissionerDeclarationHandler::OnCommissionerDeclarationMessage()");
+    CastingPlayer * targetCastingPlayer = CastingPlayer::GetTargetCastingPlayer();
+    if (targetCastingPlayer == nullptr)
+    {
+        ChipLogError(AppServer,
```

Comment:
Why are we not aborting when we find this kind of error? What are codepaths that could cause this?

Otherwise would it be not clearer to return on this error and not follow through with the rest? This would allow us to assume we have a target casting player below and not have extra checks or potential for future code to try to access it.

Reply:
I wonder if this function should be split somehow. We seem to tolerate invalid states and log errors and do several things:

- targetCastingPlayer stop connecting (tolerates casting player being nullptr)
- Commissioning window closing (gated by passcode)
- mCmmissionerDeclarationCallback_ (tolerates it being nullptr)

Overall it seems brittle in terms of tolerating everything not set and I am unclear why we moved the targetCastingPlayer with its nullcheck far away from the usage - this seems to make it more likely to mistake it for being non-null.

Reply:
Hey Andrei.

Moving the targetCastingPlayer back to where it it used in `if (cd.GetCancelPasscode())`.

I moved `CastingPlayer * targetCastingPlayer = CastingPlayer::GetTargetCastingPlayer();` and `targetCastingPlayer == nullptr` check up, since I had some additional code (removed from PR)  in `if (cd.GetCommissionerPasscode())` to check if the targetCastingPlayer is in the connecting state and with the CastingPlayer/Commissioner-Generated feature. I removed that additional check but then forgot to move targetCastingPlayer back down into `if (cd.GetCancelPasscode())` where it is used. 

This additional state check is not needed since per the Matter spec section 7.5. UDC with Commissioner-generated Passcode, Step # 9:

9. Commissioner sends CommissionerDeclaration with PasscodeDialogDisplayed and CommissionerPasscode set to true. If a QR code is also displayed, then QRCodeDisplayed is also set to true.

https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/e02a0dfcee3284f6f4b30321ffc7202290ab5bb5/src/rendezvous/UserDirectedCommissioning.adoc#75-udc-with-commissioner-generated-passcode

So we can assume we're in UDC with Commissioner-generated Passcode given the CommissionerPasscode flag.

The spec also calls out PasscodeDialogDisplayed in Step # 9 which doesn't make sense.  PasscodeDialogDisplayed is only used for the "Coordinate PIN Dialogs" feature. I think this step needs to be corrected. I have it in my to do's to submit a correction request for this. 

We also don't abort when targetCastingPlayer is null since we still want to pass the CommissionerDeclaration to the client if the Client provided the mCmmissionerDeclarationCallback_

Per the Matter User Directed Commissioning (UDC) spec: "There is no state or session associated with UDC protocol messages, allowing the recipient to choose to support messages received from different sources at the same time, or choose to ignore messages received while the processing of an earlier message has not completed."

We're opting to pass these messages to the Client for them to decide how to proceed based on the CommissionerDeclaration error code (there are 16 possible error codes).